packages/core/src/components/DatePicker/DatePicker.tsx

```diff
@@ -48,7 +48,11 @@ const Component: FC<DatePickerProps> = memo(
             isErrorPresent = useMemo(() => !!errorText || !!builtInErrorMessage, [errorText, builtInErrorMessage]);
 
         useEffect(() => {
-            date && setTextValue(format(date, displayFormat!).replace(new RegExp('\\/|\\-', 'g'), ' $& '));
+            if (date === null) {
+                setTextValue('');
+            } else if (date) {
+                setTextValue(format(date, displayFormat!).replace(new RegExp('\\/|\\-', 'g'), ' $& '));
+            }
```

Comment:
If value is changed and set to `undefined` which also produces same issue, hence it can also be handled with

```suggestion
            if (!date) {
                setTextValue('');
            } else {
                setTextValue(format(date, displayFormat!).replace(new RegExp('\\/|\\-', 'g'), ' $& '));
            }
```

Reply:
Agree with you but:

1. The `!date` means any nullish value, so it could be a few things like ``, false, `0`, Nan, so it is too broader definition to add.
2. The expectation is that we are using `Typescript` properly, sending props that are not matching the definition, and the definition for value `value: Date | string | null;`. Ideally the cod using the component should guard rendering this component if any other type of `value` is provided.

Reply:
Agree with @lucian-s-medly